tests/unit/array/mixins/test_match.py

```diff
@@ -68,6 +69,23 @@ def doc_lists_to_doc_arrays(doc_lists, *args, **kwargs):
     return D1, D2
 
 
+@pytest.mark.parametrize(
+    'storage, config', [('weaviate', WeaviateConfig(3)), ('pqlite', {'n_dim': 3})]
+)
+@pytest.mark.parametrize('limit', [1, 2, 3])
+def test_match(storage, config, doc_lists, limit, start_weaviate):
+    D1, D2 = doc_lists_to_doc_arrays(doc_lists)
+
+    if config:
+        da = DocumentArray(D2, storage=storage, config=config)
+    else:
+        da = DocumentArray(D2, storage=storage)
+
+    D1.match(da, limit=limit)
+    for m in D1[:, 'matches']:
+        assert len(m) == limit
```

Comment:
does it make sense to have further checks on the order ?

Reply:
done